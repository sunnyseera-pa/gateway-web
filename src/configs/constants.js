export const DISPLAY_DATE_STANDARD = 'd MMMM yyyy';
export const TEXTAREA_ROWS = 8;

export const DATASETS_STATUS_ACTIVE = 'active,draft';
export const STATUS_INREVIEW = 'inReview';
export const STATUS_REJECTED = 'rejected';
export const STATUS_ARCHIVE = 'archive';
