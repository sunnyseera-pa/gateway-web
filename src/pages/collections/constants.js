export const DATASET = 'dataset';
export const PERSON = 'person';
export const RELEVANCE = 'relevance';
export const POPULARITY = 'popularity';
export const RECENTLYADDED = 'recentlyadded';
export const RESOURCES = 'resources';
export const METADATA = 'metadata';
export const MAXRESULTS = 24;
