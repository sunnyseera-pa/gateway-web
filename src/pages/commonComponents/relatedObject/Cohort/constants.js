const tab_url = '/search?search=&tab=Cohorts';
export const cohort = {
	TAB: 'Cohort',
	FC: {
		parentKey: 'cohortinclusionexclusion',
		filter: 'cohortFilterCriteriaSelected',
		tagType: 'tag',
		url: `${tab_url}&cohortinclusionexclusion=`,
	},
};
