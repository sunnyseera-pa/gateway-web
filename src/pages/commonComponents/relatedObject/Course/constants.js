const tab_url = '/search?search=&tab=Courses';
export const course = {
	TAB: 'Course',
	AWARDS: {
		parentKey: 'courseaward',
		filter: 'courseaward',
		tagType: 'tag',
		url: `${tab_url}&courseaward=`,
	},
	DOMAINS: {
		parentKey: 'coursedomains',
		filter: 'courseDomainsSelected',
		tagType: 'tag',
		url: `${tab_url}&coursedomains=`,
	},
};
