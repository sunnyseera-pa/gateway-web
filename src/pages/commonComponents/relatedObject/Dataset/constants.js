const tab_url = '/search?search=&tab=Datasets';
export const dataset = {
	TAB: 'Dataset',
	PHENOTYPES: {
		parentKey: 'phenotypes',
		filter: 'phenotypes',
		tagType: 'phenotype',
		url: `${tab_url}&phenotypes=`,
	},
	FEATURES: {
		parentKey: 'datasetfeatures',
		filter: 'datasetfeatures',
		tagType: 'tag',
		url: `${tab_url}&datasetfeatures=`,
	},
};
