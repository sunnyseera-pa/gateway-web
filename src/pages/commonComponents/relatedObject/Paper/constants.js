const tab_url = '/search?search=&tab=Paper';
export const paper = {
    TAB: 'Paper',
    FEATURES: {
        parentKey: 'paperfeatures',
        filter: 'paperFeaturesSelected',
        tagType: 'tag',
        url: `${tab_url}&paperfeatures=`,
    },
    TOPICS: {
        parentKey: 'papertopics',
        filter: 'paperTopicsSelected',
        tagType: 'tag',
        url: `${tab_url}&papertopics=`,
    },
};
