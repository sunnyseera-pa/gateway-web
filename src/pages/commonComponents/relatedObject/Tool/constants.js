const tab_url = '/search?search=&tab=Tools';
export const tool = {
	TAB: 'Datasets',
	FEATURES: {
		parentKey: 'toolfeatures',
		filter: 'toolFeaturesSelected',
		tagType: 'tag',
		url: `${tab_url}&toolfeatures=`,
	},
	TOPICS: {
		parentKey: 'tooltopics',
		filter: 'toolTopicsSelected',
		tagType: 'tag',
		url: `${tab_url}&tooltopics=`,
	},
	CATEGORIES: {
		parentKey: 'toolcategories',
		filter: 'toolCategoriesSelected',
		tagType: 'tag',
		url: `${tab_url}&toolcategories=`,
	},
	PL: {
		parentKey: 'toolprogrammingLanguage',
		filter: 'toolProgrammingLanguageSelected',
		tagType: 'version',
		url: `${tab_url}&toolprogrammingLanguage=`,
	},
};
