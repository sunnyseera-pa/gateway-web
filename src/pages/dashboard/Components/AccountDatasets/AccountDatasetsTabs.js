import React from 'react';
import { Col, Row, Tab, Tabs } from 'react-bootstrap';
import '../../Dashboard.scss';

const AccountDatasetsTabs = ({ counts = { inReview: 0, active: 0, rejected: 0, archive: 0 }, team, onSelectTab, activeKey }) => {
	return (
		<div className='tabsBackground mb-0'>
			<Row>
				<Col sm={12} lg={12}>
					{team === 'admin' ? (
						<Tabs className='dataAccessTabs gray700-13' activeKey={activeKey} onSelect={onSelectTab}>
							<Tab eventKey='inReview' title={'Pending approval (' + counts.inReview + ')'}>
								{' '}
							</Tab>
						</Tabs>
					) : (
						<Tabs className='dataAccessTabs gray700-13' activeKey={activeKey} onSelect={onSelectTab}>
							<Tab eventKey='active' title={'Active (' + counts.active + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='inReview' title={'Pending approval (' + counts.inReview + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='rejected' title={'Rejected (' + counts.rejected + ')'}>
								{' '}
							</Tab>
							<Tab eventKey='archive' title={'Archived (' + counts.archive + ')'}>
								{' '}
							</Tab>
						</Tabs>
					)}
				</Col>
			</Row>
		</div>
	);
};

export default AccountDatasetsTabs;
