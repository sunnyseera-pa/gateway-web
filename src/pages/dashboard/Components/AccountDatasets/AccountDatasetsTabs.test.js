import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import AccountDatasetsTabs from './AccountDatasetsTabs';
import { STATUS_ARCHIVE } from '../../../../configs/constants';

const props = {
    counts: { inReview: 1, active: 2, rejected: 3, archive: 4, ['active,draft']: 5 },
    team: 'admin',
    onSelectTab: jest.fn(),
    activeKey: STATUS_INREVIEW,
};

jest.mock('react', () => {
    const React = jest.requireActual('react');
    React.Suspense = ({ children }) => children;
    return React;
});

let wrapper;

describe('Given the AccountDatasetsTabs component', () => {
    describe('When it is rendered and team is admin', () => {
        beforeAll(() => {
            wrapper = render(<AccountDatasetsTabs {...props} />, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Only Pending approval tab should be rendered', () => {
            expect(wrapper.queryByText('Pending approval (1)')).toBeTruthy();
            expect(wrapper.queryByText('Active (5)')).toBeNull();
        });
    });

    describe('when team is not admin', () => {
        beforeAll(() => {
            wrapper.rerender(<AccountDatasetsTabs {...props} team='manager' />, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        it('Active tab should be rendered', async () => {
            expect(screen.queryByText('Active (5)')).toBeTruthy();
        });

        it('Rejected tab should be rendered', async () => {
            expect(screen.queryByText('Rejected (3)')).toBeTruthy();
        });

        it('Pending approval tab should be rendered', async () => {
            expect(screen.queryByText('Pending approval (1)')).toBeTruthy();
        });

        it('Archived tab should be rendered', async () => {
            expect(screen.queryByText('Archived (4)')).toBeTruthy();
        });

        it('onSelect Tab', async () => {
            fireEvent.click(screen.queryByText('Archived (4)'));
            expect(props.onSelectTab.mock.calls.length).toBe(1);
            expect(props.onSelectTab.mock.calls[0][0]).toBe(STATUS_ARCHIVE);
        });
    });
});
