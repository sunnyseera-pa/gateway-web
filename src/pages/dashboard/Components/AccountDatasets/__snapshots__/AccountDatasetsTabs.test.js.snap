// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDatasetsTabs component When it is rendered and team is admin Then matches the previous snapshot 1`] = `
<div>
  <div
    class="tabsBackground mb-0"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-12 col-sm-12"
      >
        <nav
          class="dataAccessTabs gray700-13 nav nav-tabs"
          role="tablist"
        >
          <a
            aria-selected="true"
            class="nav-item nav-link active"
            data-rb-event-key="inReview"
            href="#"
            role="tab"
          >
            Pending approval (1)
          </a>
        </nav>
        <div
          class="tab-content"
        >
          <div
            aria-hidden="false"
            class="fade tab-pane active show"
            role="tabpanel"
          >
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given the AccountDatasetsTabs component when team is not admin Then matches the previous snapshot 1`] = `
<div>
  <div
    class="tabsBackground mb-0"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-12 col-sm-12"
      >
        <nav
          class="dataAccessTabs gray700-13 nav nav-tabs"
          role="tablist"
        >
          <a
            aria-selected="false"
            class="nav-item nav-link"
            data-rb-event-key="active,draft"
            href="#"
            role="tab"
          >
            Active (5)
          </a>
          <a
            aria-selected="true"
            class="nav-item nav-link active"
            data-rb-event-key="inReview"
            href="#"
            role="tab"
          >
            Pending approval (1)
          </a>
          <a
            aria-selected="false"
            class="nav-item nav-link"
            data-rb-event-key="rejected"
            href="#"
            role="tab"
          >
            Rejected (3)
          </a>
          <a
            aria-selected="false"
            class="nav-item nav-link"
            data-rb-event-key="archive"
            href="#"
            role="tab"
          >
            Archived (4)
          </a>
        </nav>
        <div
          class="tab-content"
        >
          <div
            aria-hidden="true"
            class="fade tab-pane"
            role="tabpanel"
          >
             
          </div>
          <div
            aria-hidden="false"
            class="fade tab-pane active show"
            role="tabpanel"
          >
             
          </div>
          <div
            aria-hidden="true"
            class="fade tab-pane"
            role="tabpanel"
          >
             
          </div>
          <div
            aria-hidden="true"
            class="fade tab-pane"
            role="tabpanel"
          >
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
